### PR TITLE
Add DeleteForStudentsAndGroupsCommand and relevant test cases

### PR DESCRIPTION
**Pull request by zannloo**
Parser has been updated for this command.

### CODE REVIEW

**Review by mingyuanc — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/newcommands/DeleteForStudentsAndGroupsCommand.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    public CommandResult execute(State state) throws CommandException {
+        AbsolutePath currPath = state.getCurrPath();
+        Root root = state.getRoot();
+        CommandResult returnStatement;
+        try {
+            requireAllNonNull(currPath, root);
+            AbsolutePath absolutePath = currPath.resolve(path);
+
+            if (absolutePath.isStudentDirectory()) {
+                ChildOperation<Student> target = StateManager.groupChildOperation(root, absolutePath);
```
Comment (mingyuanc) [inline src/main/java/seedu/address/logic/newcommands/DeleteForStudentsAndGroupsCommand.java:64]: Please change to the new methods

**Review by mingyuanc — CHANGES_REQUESTED** [review]

**Review by mingyuanc — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/newcommandparser/DeleteForStudentsAndGroupsCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                ArgumentTokenizer.tokenize(args);
+
+        if (argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(
+                    MESSAGE_INVALID_COMMAND_FORMAT, DeleteForStudentsAndGroupsCommand.MESSAGE_USAGE));
+        }
+        RelativePath path = ParserUtil.parsePath(argMultimap.getPreamble());
+        if (!path.isGroupDirectory() && !path.isStudentDirectory()) {
+            throw new ParseException(INVALID_PATH_MESSAGE);
+        }
```
Comment (NereusWB922) [inline src/main/java/seedu/address/logic/parser/newcommandparser/DeleteForStudentsAndGroupsCommandParser.java:34]: Should check this in command execute.
Consider the situation: relative path = ../ , u won't know what is this refer to without current path

**Review by NereusWB922 — CHANGES_REQUESTED** [review]
Need changes for Parser

**Review by NgChunMan — APPROVED** [review]
LGTM.

**Review by NereusWB922 — APPROVED** [review]

**Pull request merged**